code/game/objects/items/surgery_tray.dm

```diff
@@ -0,0 +1,334 @@
+		var/obj/item/circular_saw/other_saw = locate() in contents
+		circular_saw = other_saw || null
+	else if(gone == hemostat)
+		var/obj/item/hemostat/other_hemostat = locate() in contents
+		hemostat = other_hemostat || null
+	else if(gone == razor)
+		var/obj/item/razor/other_razor = locate() in contents
+		razor = other_razor || null
+	else if(gone == surgical_drill)
+		var/obj/item/surgicaldrill/other_drill = locate() in contents
+		surgical_drill = other_drill || null
+	else if(gone == retractor)
+		var/obj/item/retractor/other_retractor = locate() in contents
+		retractor = other_retractor || null
+	else if(gone == bone_gel)
+		var/obj/item/stack/medical/bone_gel/other_gel = locate() in contents
+		bone_gel = other_gel || null
+	else if(gone == surgical_tape)
+		var/obj/item/stack/sticky_tape/other_tape = locate() in contents
+		surgical_tape = other_tape || null
+	else if(gone == surgical_drapes)
+		var/obj/item/surgical_drapes/other_drapes = locate() in contents
+		surgical_drapes = other_drapes || null
+
+/obj/item/surgery_tray/proc/PopulateContents()
+	blood_filter = new(src)
+	bonesetter = new(src)
+	razor = new(src)
+	if(!cautery)
+		cautery = new(src)
+	if(!hemostat)
+		hemostat = new(src)
+	if(!retractor)
+		retractor = new(src)
+	if(!scalpel)
+		scalpel = new(src)
+	circular_saw = new(src)
+	bone_gel = new(src)
+	surgical_tape = new(src)
+	surgical_drapes = new(src)
+	surgical_drill = new(src)
+	var/static/list/items_inside = list(
+		/obj/item/clothing/mask/surgical = 1,
+	)
+	generate_items_inside(items_inside, src)
+
+///Sets the surgery tray's deployment state. Silent if user is null.
+/obj/item/surgery_tray/proc/set_tray_mode(new_mode, mob/user)
+	tray_toggled = new_mode
+	density = !tray_toggled
+
+	if(user)
+		user.visible_message(span_notice("[user] [tray_toggled ? "retracts" : "extends"] [src]'s wheels."), span_notice("You [tray_toggled ? "retract" : "extend"] [src]'s wheels."))
+
+	if(tray_toggled)
+		interaction_flags_item |= INTERACT_ITEM_ATTACK_HAND_PICKUP
+		pass_flags |= PASSTABLE
+	else
+		interaction_flags_item &= ~INTERACT_ITEM_ATTACK_HAND_PICKUP
+		pass_flags &= ~PASSTABLE
+
+	update_appearance()
+
+/obj/item/surgery_tray/equipped(mob/user, slot, initial)
+	. = ..()
+	if(!tray_toggled)
+		set_tray_mode(TRUE, user)
+
+/obj/item/surgery_tray/attack_self(mob/user, modifiers)
+	. = ..()
+	if(.)
+		return
+	var/turf/open/placement_turf = get_turf(user)
+	if(isgroundlessturf(placement_turf) || isclosedturf(placement_turf))
+		balloon_alert(user, "can't deploy!")
+		return TRUE
+	if(!user.transferItemToLoc(src, placement_turf))
+		balloon_alert(user, "tray stuck!")
+		return TRUE
+	set_tray_mode(FALSE, user)
+	return
+
+/obj/item/surgery_tray/attack_hand(mob/living/user)
+	if(!user.can_perform_action(src, NEED_HANDS))
+		return ..()
+	var/obj/item/grabbies = pick(contents)
+	if(grabbies)
+		atom_storage.remove_single(user, grabbies, drop_location())
+		user.put_in_hands(grabbies)
+	return TRUE
```

Comment:
didn't touch this because it was unrelated but i wanna ask- why does LMB grab a random item from the tray?
Shouldn't it be the inverse, that LMB is the default behavior we want to be done on click, and RMB is the secondary?

Reply:
rmb to open is consistent with other storage objects, lmb has the bonus of being the default key to put an item back in the tray as well. 
when i tried with alt rmb, it was somewhat out of the way, so it was somewhat unlikely to be pressed, and spamming it would just dump all the tools of the tray out one by one. left click on the other hand, repeatedly swaps to a random tool or item when you spam click it.
i wouldn't be too put out if you swapped it or asked me to, but i did think on it quite a bit before settling on this. if someone changes it in a future PR i also probably won't mind much

Reply:
just swapping it would probably be fine though that was a lil long winded

Reply:
LMB is consistent with other storage objects too though? LMB should be the default function which for this you would want to pick the item you desired, not take a random one

Reply:
okay, i'll try and swap them sometime today

Reply:
> LMB is consistent with other storage objects too though? LMB should be the default function which for this you would want to pick the item you desired, not take a random one

turns out LMB is only the storage interact key if it's already in your inventory, and overriding the RMB behavior it inherits from storage objects would be trickier (and messier) than i'd like. if you have any tips on how i'd implement it feel free to ping me in discord and we can try rigging it up, though

Reply:
Oh, if that's the case, at least allow the player to make the choice through a (persistent) radial menu instead of picking random tool from the bunch. That'd definitely prove more useful.